improver_tests/utilities/test_cube_checker.py

```diff
@@ -182,32 +187,37 @@ def test_coord_promotion_missing_scalar(self):
         """Test case in which a scalar coordinate has been lost from new_cube,
         meaning the cube undergoing checking ends up with different dimension
         coordinates to the progenitor cube. This raises an error."""
-        cube = set_up_cube()
-        new_cube = iris.util.squeeze(cube)
+        new_cube = iris.util.squeeze(self.cube)
         new_cube.remove_coord("realization")
```

Comment:
Why is the scalar realization coordinate removed?  The error raised here should be raised regardless, because the dimensions don't match.

Reply:
This function checks to see if a dimension coordinate has been demoted to a scalar by comparing a cube against a reference cube. If it finds a scalar that is a dimension on the reference, it promotes it, thus making the dimensionality of the two cubes match once again. In this case the scalar is removed, such that it is not found to be promoted. After the call to the function, the dimension don't match and the missing coordinate is not included in an exemption list. Hence an exception. This is the purpose of this test.